app/waf/src/services/bigip.service.ts

```diff
@@ -196,6 +196,18 @@ export class BigIpManager {
     );
   }
 
+  async checkPartition(partition: string): Promise<boolean> {
```

Comment:
I think the codes in bigip.service.ts is only used for retrieving information from bigip, has no responsibility of check...
So returning the gotten information back and left the upper caller to do the check is reasonable.

Reply:
Let us discuss this further. In my opinion, checkPartition is the function used to fit in your design.

Thanks.

Reply:
resolved and moved the check logic outside.